smithy-cli/src/test/java/software/amazon/smithy/cli/commands/BuildCommandTest.java

```diff
@@ -52,6 +52,16 @@ public void dumpsOutValidationErrorsAndFails() throws Exception {
         assertThat(result.stderr(), containsString("FAILURE"));
     }
 
+    @Test
+    public void allowsUnknownTraitWithFlag() throws Exception {
+        String model = Paths.get(getClass().getResource("unknown-trait.smithy").toURI()).toString();
+        CliUtils.Result result = CliUtils.runSmithy("build", "--allow-unknown-traits", model);
```

Comment:
This is similar to `dumpsOutValidationErrorsAndFails` above, but with the `--allow-unknown-traits` option making it succeed.